src/sentry/api/endpoints/issues/related_issues.py

```diff
@@ -24,13 +25,26 @@ class RelatedIssuesEndpoint(GroupEndpoint):
     }
 
     # We get a Group object since the endpoint is /issues/{issue_id}/related-issues
-    def get(self, _: Request, group: Group) -> Response:
+    def get(self, request: Request, group: Group) -> Response:
         """
-        Retrieve related issues for an Issue
+        Retrieve related issues for a Group
         ````````````````````````````````````
         Related issues can be based on the same root cause or trace connected.
 
-        :pparam string group_id: the ID of the issue
+        :pparam Request request: the request object
+        :pparam Group group: the group object
         """
-        related_issues = find_related_issues(group)
+        # The type of related issues to retrieve. Can be either `same_root_cause` or `trace_connected`.
+        related_type = request.query_params.get("type")
+        related_issues: list[dict[str, str | list[int] | dict[str, str]]] = []
+
+        if related_type == "same_root_cause":
+            data, meta = same_root_cause_analysis(group)
+            related_issues = [{"type": related_type, "data": data, "meta": meta}]
+        elif related_type == "trace_connected":
+            data, meta = trace_connected_analysis(group)
+            related_issues = [{"type": related_type, "data": data, "meta": meta}]
+        else:
+            # XXX: We will be deprecating this approach soon
+            related_issues = find_related_issues(group)
         return Response({"data": [related_set for related_set in related_issues]})
```

Comment:
This return type ends up feeling a little weird once the deprecated case is removed when the result will only ever contain type. Thoughts on removing the outer array for these new responses?

ex: `{"type": "same_root_cause", "data": [5], "meta": {}}` instead of the current `[{"type": "same_root_cause", "data": [5], "meta": {}}]`

Reply:
You got a good eye Mr. Duncan!
When dealing with the follow up UI changes I could already see how non-sensical it is to keep the old shape.